### ISSUE TITLE
Investigate using GitHub Actions (vs Travis and/or BuildBot)

### ISSUE THREAD

**Issue by steven-johnson** [halide/Halide#4116]
Per https://github.blog/2019-08-08-github-actions-now-supports-ci-cd, this looks like (at least) a replacement for Travis; it looks like it might also allow for using self-hosted builders, in which case we might be able to also replace BuildBot usage and keep our custom buildbot machines for now. Someone should investigate.

**Comment by alexreinking** [comment]
@steven-johnson -- any updates on this?

Some relevant PRs:

#4792 
#4767 
#4680 
#4657 
#4667 
#4618 

Some relevant issues

#4752 
#4754 
#4753

**Comment by steven-johnson** [comment]
Yeah, this should still stay open. Unfortunately GHA still doesn't have the features needed to make it a good replacement for our Buildbots; in particular, there still isn't a good way to throttle redundant jobs (e.g., if a PR is opened and then one or two oopsie-fixes are pushed, there isn't a good way to collapse those into a single build or to kill the redundant ones).

**Comment by abadams** [comment]
> [!NOTE]
> **Automated triage:** this issue was flagged by Claude as likely safe to close, for the
> reason below. The reasoning was checked against the current source, and a second pass
> tried and failed to refute it, but no human verified this specific text.
> **If the reason is wrong, please reopen the issue and say so** — some fraction of these
> are expected to be wrong.

**Reason (CLOSE_FIXED):** The investigation happened and GitHub Actions was adopted: Travis is gone and CI for Linux, macOS, Windows, ARM Linux, the make build, pip packaging, clang-tidy and pre-commit all run on GHA. The blocker steven-johnson named in 2020 (no way to cancel redundant jobs) was fixed by GHA's concurrency groups, which the workflows now use.

**Evidence:** commit 7bedfb095 "Drop Travis testing" (no .travis.yml in tree); .github/workflows/{testing-linux,testing-macos,testing-windows,testing-arm-linux,testing-make,testing-ecosystem,pip,clang-tidy,pre-commit}.yml; .github/workflows/testing-linux.yml:23-25 `concurrency: group: ... cancel-in-progress: true`

Closing: this investigation concluded long ago — Travis was removed (7bedfb095) and CI now runs on GitHub Actions across Linux/macOS/Windows/ARM, with the redundant-job concern resolved by `concurrency: cancel-in-progress` (see .github/workflows/testing-linux.yml).
